File: solutions/1381/1381-kir3i.cpp
class CustomStack {
public:
    int size = 0;
    int maxSize;
    vector<int> data;
    CustomStack(int maxSize) {
        this->maxSize = maxSize;
    }
    
    void push(int x) {
        if (size == maxSize)
            return;
        data.push_back(x);
        size++;
    }
    
    int pop() {
        if (size == 0)
            return -1;
        size--;
        int rtn = data.back();
        data.pop_back();
        return rtn;
    }
    
    void increment(int k, int val) {
        if (size <= k)
            for(int i=0; i<size; i++)
                data[i] += val;
        else
            for(int i=0; i<k; i++)
                data[i] += val;
    }
};

File: solutions/1381/1381-kir3i.java
class CustomStack {
    public int top = 0;
    public int maxSize;
    public int[] data;
    
    public CustomStack(int maxSize) {
        this.maxSize = maxSize;
        data = new int[maxSize];
    }
    
    public void push(int x) {
        if (top == maxSize)
            return;
        data[top++] = x;
    }
    
    public int pop() {
        if (top == 0)
            return -1;
        return data[--top];
    }
    
    public void increment(int k, int val) {
        int sz = k<top? k: top;
        for(int i=0; i<sz; i++)
            data[i] += val;
    }
}
